fearless-utils/src/main/java/jp/co/soramitsu/fearless_utils/runtime/metadata/module/RuntimeMetadataModule.kt

```diff
@@ -0,0 +1,283 @@
+import jp.co.soramitsu.fearless_utils.runtime.metadata.DoubleMapSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.ErrorMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.EventMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.FunctionArgumentMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.FunctionMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.MapSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.ModuleConstantMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.NMapSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.StorageEntryMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.StorageEntryModifier
+import jp.co.soramitsu.fearless_utils.runtime.metadata.StorageHasher
+import jp.co.soramitsu.fearless_utils.runtime.metadata.StorageMetadataSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.WithName
+import jp.co.soramitsu.fearless_utils.runtime.metadata.groupByName
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.MapTypeV14
+import jp.co.soramitsu.fearless_utils.scale.EncodableStruct
+import java.math.BigInteger
+
+class Module(
+    override val name: String,
+    val storage: Storage?,
+    val calls: Map<String, MetadataFunction>?,
+    val events: Map<String, Event>?,
+    val constants: Map<String, Constant>,
+    val errors: Map<String, Error>,
+    val index: BigInteger
+) : WithName
+
+class Storage(
+    val prefix: String,
+    val entries: Map<String, StorageEntry>
+) {
+    constructor(
+        typeRegistry: TypeRegistry,
+        struct: EncodableStruct<StorageMetadataSchema>,
+        moduleName: String
+    ) : this(
+        prefix = struct[StorageMetadataSchema.prefix],
+        entries = struct[StorageMetadataSchema.entries].map {
+            StorageEntry(
+                typeRegistry,
+                it,
+                moduleName
+            )
+        }
+            .groupByName()
+    )
+
+    operator fun get(entry: String) = entries[entry]
+}
+
+class StorageEntry(
+    override val name: String,
+    val modifier: StorageEntryModifier,
+    val type: StorageEntryType,
+    val default: ByteArray,
+    val documentation: List<String>,
+    val moduleName: String
+) : WithName {
+
+    constructor(
+        typeRegistry: TypeRegistry,
+        struct: EncodableStruct<StorageEntryMetadataSchema>,
+        moduleName: String
+    ) : this(
+        name = struct[StorageEntryMetadataSchema.name],
+        modifier = struct[StorageEntryMetadataSchema.modifier],
+        type = StorageEntryType.from(typeRegistry, struct[StorageEntryMetadataSchema.type]),
+        default = struct[StorageEntryMetadataSchema.default],
+        documentation = struct[StorageEntryMetadataSchema.documentation],
+        moduleName = moduleName
+    )
+}
+
+sealed class StorageEntryType(
+    val value: Type<*>?
+) {
+
```

Comment:
I suggest to unify v13 and v14 here and only have Plain and NMap, since both versions generalizes fine to it